client/components/Assignments/AssignmentEditor.jsx

```diff
@@ -195,7 +195,7 @@ export class AssignmentEditorComponent extends React.Component {
                     label={gettext('User')}
                     value={this.state.user}
                     onChange={this.onUserChange}
-                    users={this.state.filteredUsers}
+                    users={this.state.filteredUsers.filter((u) => u.is_active && u.is_enabled)}
```

Comment:
Remove the filtering of users here, simply pass through `this.state.filteredUsers`